### PR TITLE
Use os.path.sep instead of hardcoding /

### PR DESCRIPTION
**Pull request by nijave**
##### SUMMARY
Use Python's built in path separator constant in setup

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
core

### CODE REVIEW

**Comment by jillr** [comment]
Is there a bug or use case that this is addressing?

**Comment by nijave** [comment]
> Is there a bug or use case that this is addressing?

Yeah, there's some details in the commit message:
This is the only change needed for ansible 2.9.1 to be installable on
Windows.

I know Windows isn't officially supported but being able to
install is still useful since it will end up in Python's default path
making module development easier

**Review by jamescassell — APPROVED** [review]
shipit

File: setup.py [hunk 1]
```diff
@@ -50,22 +50,22 @@ def _find_symlinks(topdir, extension=''):
             filepath = os.path.join(base_path, filename)
             if os.path.islink(filepath) and filename.endswith(extension):
                 target = os.readlink(filepath)
-                if target.startswith('/'):
+                if target.startswith(os.path.sep):
```
Comment (felixfontein) [inline setup.py:53]: Shouldn't this be:
```suggestion
                if os.path.isabs(target):
```

File: setup.py [hunk 2]
```diff
@@ -50,22 +50,22 @@ def _find_symlinks(topdir, extension=''):
...
                     # We do not support absolute symlinks at all
                     continue
 
                 if os.path.dirname(target) == '':
                     link = filepath[len(topdir):]
-                    if link.startswith('/'):
+                    if link.startswith(os.path.sep):
```
Comment (felixfontein) [inline setup.py:59]:
```suggestion
                    if os.path.isabs(link):
```

File: setup.py [hunk 3]
```diff
@@ -50,22 +50,22 @@ def _find_symlinks(topdir, extension=''):
...
                     symlinks[os.path.basename(target)].append(link)
                 else:
                     # Count how many directory levels from the topdir we are
-                    levels_deep = os.path.dirname(filepath).count('/')
+                    levels_deep = os.path.dirname(filepath).count(os.path.sep)
 
                     # Count the number of directory levels higher we walk up the tree in target
                     target_depth = 0
-                    for path_component in target.split('/'):
+                    for path_component in target.split(os.path.sep):
```
Comment (felixfontein) [inline setup.py:68]: This might not work well for Windows paths with drive letters.

File: setup.py [hunk 4]
```diff
@@ -76,7 +76,7 @@ def _find_symlinks(topdir, extension=''):
                     else:
                         # If we managed to stay within the tree, store the symlink
                         link = filepath[len(topdir):]
-                        if link.startswith('/'):
+                        if link.startswith(os.path.sep):
```
Comment (felixfontein) [inline setup.py:79]: I guess this should also be `os.path.isabs(link)`, but the next line also needs to be adjusted.

**Comment by sivel** [comment]
I'm not in favor of this. Enforcing it will become difficult, and I think it sends the wrong message. Changing this, adds to the misconception that we may support a windows controller. As such, more people may want to add additional changes, and we aren't prepared to support this.  Because we aren't prepared, this will be a constant losing battle.

With the above in mind, I'm going to close this.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Comment by nijave** [comment]
> I'm not in favor of this. Enforcing it will become difficult, and I think it sends the wrong message. Changing this, adds to the misconception that we may support a windows controller. As such, more people may want to add additional changes, and we aren't prepared to support this. Because we aren't prepared, this will be a constant losing battle.
> 
> With the above in mind, I'm going to close this.
> 
> If you have further questions please stop by IRC or the mailing list:
> 
>     * IRC: #ansible on irc.freenode.net
> 
>     * mailing list: https://groups.google.com/forum/#!forum/ansible-project

If that's the case then why rely on operating system-specific semantics instead of explicitly rejecting unsupported operating systems?

> adds to the misconception that we may support a windows controller.

This misconception already exists so why not explicitly block install

**Comment by jamescassell** [comment]
> I'm not in favor of this. Enforcing it will become difficult, and I think it sends the wrong message. Changing this, adds to the misconception that we may support a windows controller. As such, more people may want to add additional changes, and we aren't prepared to support this. Because we aren't prepared, this will be a constant losing battle.
> 
> With the above in mind, I'm going to close this.
> 
> If you have further questions please stop by IRC or the mailing list:
> 
>     * IRC: #ansible on irc.freenode.net
> 
>     * mailing list: https://groups.google.com/forum/#!forum/ansible-project

sounds like we need to fork ansible if we want it to work on windows...

**Comment by nitzmahone** [comment]
A blog post on this topic was waaay overdue: http://blog.rolpdog.com/2020/03/why-no-ansible-controller-for-windows.html

(more color on ML thread as well: https://groups.google.com/forum/m/#!topic/ansible-project/mC5t9kYbZew)

**Pull request closed without merging**